src/site/stages/build/plugins/create-megamenu.js

```diff
@@ -0,0 +1,49 @@
+const fs = require('fs-extra');
+const path = require('path');
+const yaml = require('js-yaml');
+
+const { applyFragments } = require('./apply-fragments');
+
+const MEGAMENU_DATA_SOURCE_FILENAME = 'megamenu/index.yml';
+const MEGAMENU_CACHE_FILENAME = 'megamenu.json';
+
+function writeToCache(buildOptions, megaMenuData) {
+  const megamenuDataCacheFileName = path.join(
+    buildOptions.cacheDirectory,
+    MEGAMENU_CACHE_FILENAME,
+  );
+  const serialized = JSON.stringify(megaMenuData, null, 4);
+
+  fs.ensureDirSync(buildOptions.cacheDirectory);
+  fs.writeFileSync(megamenuDataCacheFileName, serialized);
+}
+
+function createMegaMenu(buildOptions) {
```

Comment:
I updated the function name but left the file name alone in case we do anything more with the MegaMenu there, like link validation or using ReactDOMServer.